datafusion/physical-expr-common/src/physical_expr.rs

```diff
@@ -106,6 +106,9 @@ pub trait PhysicalExpr: Send + Sync + Display + Debug + DynEq + DynHash {
             Ok(tmp_result)
         } else if let ColumnarValue::Array(a) = tmp_result {
             scatter(selection, a.as_ref()).map(ColumnarValue::Array)
+        } else if let ColumnarValue::Scalar(ScalarValue::Boolean(v)) = &tmp_result {
+            let a = BooleanArray::from(vec![*v; tmp_batch.num_rows()]);
```

Comment:
we can probably make things faster if we avoid expanding out to a constant array here (avoid the `vec![]`).